scipy/optimize/tests/test_bracket.py

```diff
@@ -763,93 +771,98 @@ def bracket_minimum_single(xm0, xl0, xr0, xmin, xmax, factor, a):
+        args = tuple(xp.asarray(arg, dtype=xp.float64) for arg in args)
+        res = _bracket_minimum(f, xp.asarray(xm0), xl0=xl0, xr0=xr0, xmin=xmin, xmax=xmax,
+                               factor=factor, args=args, maxiter=maxiter)
 
         attrs = ['xl', 'xm', 'xr', 'fl', 'fm', 'fr', 'success', 'nfev', 'nit']
         for attr in attrs:
-            ref_attr = [getattr(ref, attr) for ref in refs]
+            ref_attr = [xp.asarray(getattr(ref, attr)) for ref in refs]
             res_attr = getattr(res, attr)
-            assert_allclose(res_attr.ravel(), ref_attr)
-            assert_equal(res_attr.shape, shape)
+            xp_assert_close(xp_ravel(res_attr, xp=xp), xp.stack(ref_attr))
+            xp_assert_equal(res_attr.shape, shape)
 
-        assert np.issubdtype(res.success.dtype, np.bool_)
+        xp_test = array_namespace(xp.asarray(1.))
+        assert res.success.dtype == xp_test.bool
         if shape:
-            assert np.all(res.success[1:-1])
-        assert np.issubdtype(res.status.dtype, np.integer)
-        assert np.issubdtype(res.nfev.dtype, np.integer)
-        assert np.issubdtype(res.nit.dtype, np.integer)
-        assert_equal(np.max(res.nit), f.count - 3)
-        self.assert_valid_bracket(res)
-        assert_allclose(res.fl, f(res.xl, *args))
-        assert_allclose(res.fm, f(res.xm, *args))
-        assert_allclose(res.fr, f(res.xr, *args))
-
-    def test_special_cases(self):
+            assert xp.all(res.success[1:-1])
+        assert res.status.dtype == xp.int32
+        assert res.nfev.dtype == xp.int32
+        assert res.nit.dtype == xp.int32
+        assert xp.max(res.nit) == f.count - 3
+        self.assert_valid_bracket(res, xp)
+        xp_assert_close(res.fl, f(res.xl, *args))
+        xp_assert_close(res.fm, f(res.xm, *args))
+        xp_assert_close(res.fr, f(res.xr, *args))
+
+    def test_special_cases(self, xp):
         # Test edge cases and other special cases.
+        xp_test = array_namespace(xp.asarray(1.))
 
         # Test that integers are not passed to `f`
         # (otherwise this would overflow)
         def f(x):
-            assert np.issubdtype(x.dtype, np.floating)
+            assert xp_test.isdtype(x.dtype, "numeric")
             return x ** 98 - 1
 
-        result = _bracket_minimum(f, -7, xr0=5)
+        result = _bracket_minimum(f, xp.asarray(-7., dtype=xp.float64), xr0=5)
         assert result.success
 
         # Test maxiter = 0. Should do nothing to bracket.
         def f(x):
             return x**2 - 10
 
-        xl0, xm0, xr0 = -3, -1, 2
+        xl0, xm0, xr0 = xp.asarray(-3.), xp.asarray(-1.), xp.asarray(2.)
         result = _bracket_minimum(f, xm0, xl0=xl0, xr0=xr0, maxiter=0)
-        assert_equal([result.xl, result.xm, result.xr], [xl0, xm0, xr0])
+        xp_assert_equal(xp.asarray([result.xl, result.xm, result.xr]),
+                        xp.asarray([xl0, xm0, xr0]))
```

Comment:
Probably should compare these separately so you don't use `xp.asarray` on the result. (One isolated case doesn't matter much, but it would not be a good habbit because we want to make sure that the result is already an array of the expected type.)